borsh-derive/src/internals/attributes/item/mod.rs

```diff
@@ -80,6 +81,16 @@ pub(crate) fn contains_use_discriminant(input: &ItemEnum) -> Result<bool, syn::E
     Ok(use_discriminant.unwrap_or(false))
 }
 
+/// Gets type of reprc attribute if it exists
+pub(crate) fn get_maybe_reprc_attribute(input: &ItemEnum) -> Option<TypePath> {
+    input
+        .attrs
+        .iter()
+        .find(|x| x.path() == REPR)?
```

Comment:
also, this behaviour should be controlled by borsh-specific item attribute, say `#[borsh(enum_repr = "i16")]`,
not by `repr`, which is used to control memory layout in program, which is, strictly speaking, independent of how it's serialized for later transfer or saving to disk (which is what serialization is usually used for). 
Say, an enum may not have `#[repr(u16)]` and have all discriminants withing `u8` range, but still have `#[borsh(enum_repr = "u16")]` and use 2 bytes for tag serialization.

Reply:
Assuming we have borsh enum_repr, how should borsh handle this

```
fn main(){
    #[borsh(enum_repr = "u16")]
    #[repr(u32)]
    enum A {
        Q = 1u8,
    }
}
```
?

Reply:
may use `u16` instead of `"u16"`? may just do `borsh(rerp=` instead of #[borsh(enum_repr` because it is on enum anyway?

Reply:
may it be like this?
```
    #[borsh(repr = u16)]
    enum A {
        Q = 1u8,
    }
```

and fail compile same way `#repr` fails because u8 was set (so borsh rerp interpreted types same way/algorithm) as native repr.

Reply:
> may it be like this?

looks good, `repr` should then check it annotates an enum item.

> fails because u8 was set

well, one would've to change 1u8 to 1 to compile with borsh attribute.

Specifying `Q = -10` would allow annotating with `#[repr(i8)]` and `#[borsh(repr = i16)]` at the same time

Reply:
> may it be like this?


Still, maybe naming it `enum_tag` will be better, rust's `repr` has other subattributes, not related to enums at all